server/src/main/scala/org/ada/server/dataaccess/dataset/SimpleInitializableCache.scala

```diff
@@ -0,0 +1,90 @@
+package org.ada.server.dataaccess.dataset
+
+import collection.mutable.{Map => MMap}
+import scala.concurrent.{Await, Future}
+import play.api.libs.concurrent.Execution.Implicits.defaultContext
+import java.util.concurrent.ConcurrentHashMap
+
+import scala.concurrent.duration._
+
+abstract class SimpleInitializableCache[ID, T](eagerInit: Boolean) {
```

Comment:
Would you mind documenting this class and the methods it exposes?

Reply:
will do

Reply:
addressed in 800fa85